yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -165,6 +166,7 @@ export class AztecNodeService implements AztecNode {
       new DataTxValidator(),
       new MetadataTxValidator(config.l1ChainId),
       new TxProofValidator(proofVerifier),
+      new DoubleSpendTxValidator(new WorldStateDB(worldStateSynchronizer.getLatest())),
```

Comment:
Why is this validator being added here?

Reply:
But isn't this PR about having `simulate` check with the node for note validity? What this change does is extend the validations the node performs to also check for double spends, which is unrelated.